src/d3a/d3a_core/sim_results/stats.py

```diff
@@ -107,6 +110,26 @@ def _get_child_data(self, area):
                         for child in area.children}
             return self.bills_results[area.name]
 
+    def _store_area_penalties(self, results_dict, area):
+        if len(area.children) > 0:
+            return
+        if isinstance(area.strategy, LoadHoursStrategy):
+            penalty_energy = sum(
+                area.strategy.energy_requirement_Wh.get(market.time_slot, 0) / 1000.0
+                for market in self._get_past_markets_from_area(area.parent, "past_markets"))
+        elif isinstance(area.strategy, PVStrategy):
+            penalty_energy = sum(
+                area.strategy.state.available_energy_kWh.get(market.time_slot, 0)
+                for market in self._get_past_markets_from_area(area.parent, "past_markets"))
+        else:
+            return
+        if "penalty_energy" not in results_dict:
+            results_dict["penalty_energy"] = 0.0
+            results_dict["penalty_cost"] = 0.0
+
+        results_dict["penalty_energy"] += penalty_energy
+        results_dict["penalty_cost"] += penalty_energy * DEVICE_PENALTY_RATE
```

Comment:
I think the bills are accumulated in Euro or Dollar not in cents. But DEVICE_PENALTY_RATE is in cents AFAIU.
Right?

Reply:
Correct! This calculation is currently returning cents, therefore I will convert it to Euro. Thanks!